### PR TITLE
refactor: migrate back to net/http

### PR DESCRIPTION
**Pull request by kpetremann**
There is no reason to keep using a third party lib since net/http introduced features we needed.

### CODE REVIEW

File: internal/api/auth/basic_auth.go [hunk 1]
```diff
@@ -74,24 +73,24 @@ func (b *BasicAuth) configureLdap(ldap *LDAPAuth) error {
 	return nil
 }
 
-func (b *BasicAuth) Wrap(next httprouter.Handle) httprouter.Handle {
+func (b *BasicAuth) Wrap(next http.HandlerFunc) http.HandlerFunc {
 	switch b.mode {
 	case noAuth:
-		return func(w http.ResponseWriter, r *http.Request, ps httprouter.Params) { next(w, r, ps) }
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) { next(w, r) })
```
Comment (ShimmerGlass) [inline internal/api/auth/basic_auth.go:79]: `http.HandlerFunc` is not needed

File: internal/api/auth/basic_auth.go [hunk 2]
```diff
@@ -74,24 +73,24 @@ func (b *BasicAuth) configureLdap(ldap *LDAPAuth) error {
...
 	case ldapMode:
 		return BasicAuthLDAP(b.ldapAuth, next)
 	default:
-		return func(w http.ResponseWriter, _ *http.Request, _ httprouter.Params) {
+		return http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
```
Comment (ShimmerGlass) [inline internal/api/auth/basic_auth.go:83]: same

File: internal/api/auth/basic_auth.go [hunk 3]
```diff
@@ -74,24 +73,24 @@ func (b *BasicAuth) configureLdap(ldap *LDAPAuth) error {
...
+		})
 	}
 }
 
 // BasicAuthLDAP is a middleware wrapping the target HTTP HandlerFunc.
 // It retrieves BasicAuth credentials and authenticate against LDAP.
-func BasicAuthLDAP(ldapAuth *LDAPAuth, next httprouter.Handle) httprouter.Handle {
-	return func(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+func BasicAuthLDAP(ldapAuth *LDAPAuth, next http.HandlerFunc) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```
Comment (ShimmerGlass) [inline internal/api/auth/basic_auth.go:93]: same

File: internal/api/auth/basic_auth.go (hunk 1 above)
Comment (kpetremann) [reply internal/api/auth/basic_auth.go]: done

**Review by arn-alt — APPROVED** [review]

**Pull request merged**